README.md

```diff
@@ -16,29 +34,254 @@ Karpenter improves the efficiency and cost of running workloads on Kubernetes cl
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+* [Helm](https://helm.sh/docs/intro/install/)
+* [jq](https://stedolan.github.io/jq/) - used by some of the scripts below
+* [yq](https://mikefarah.gitbook.io/yq/) - used by some of the scripts below
+
+### Create a cluster
+
+Create a new AKS cluster with the required configuration, and ready to run Karpenter using worload identity.
+
+> Note: You can use `hack/deploy/create-cluster.sh <cluster-name> <resource-group>` to automate the following steps. 
+
+Set environment variables:
+
+```bash
+export CLUSTER_NAME=karpenter
+export RG=karpenter
+export LOCATION=eastus
+export KARPENTER_NAMESPACE=kube-system
+
+```
+
+Create the resource group:
+
+```bash
+az group create --name ${RG} --location ${LOCATION}
+```
+
+Create the workload MSI that is the backing for the karpenter pod auth:
+
+```bash
+az identity create --name karpentermsi --resource-group "${RG}" --location "${LOCATION}"
+```
+
+Create AKS cluster compatible with Karpenter, and with the workload identity enabled:
+
+```bash
+az aks create \
+  --name "${CLUSTER_NAME}" --resource-group "${RG}" \
+  --node-count 3 --generate-ssh-keys \
+  --network-plugin azure --network-plugin-mode overlay --network-dataplane cilium \
+  --enable-managed-identity \
+  --enable-oidc-issuer --enable-workload-identity \
+  -o none
+az aks get-credentials --name "${CLUSTER_NAME}" --resource-group "${RG}" --overwrite-existing
+```
+
+Create federated credential linked to the karpenter service account for auth usage:
+
+```bash
+AKS_OIDC_ISSUER=$(az aks show -n "${CLUSTER_NAME}" -g "${RG}" --query "oidcIssuerProfile.issuerUrl" -otsv)
+az identity federated-credential create --name KARPENTER_FID --identity-name karpentermsi --resource-group "${RG}" \
+  --issuer "${AKS_OIDC_ISSUER}" \
+  --subject system:serviceaccount:${KARPENTER_NAMESPACE}:karpenter-sa \
+  --audience api://AzureADTokenExchange
+```
+
+Create role assignments to let Karpenter manage VMs and Network resources:
+
+```bash
+KARPENTER_USER_ASSIGNED_CLIENT_ID=$(az identity show --resource-group "${RG}" --name karpentermsi --query 'principalId' -otsv)
+RG_MC=$(az aks show --name "$CLUSTER_NAME" --resource-group "$RG" | jq -r ".nodeResourceGroup")
+RG_MC_RES=$(az group show --name "${RG_MC}" --query "id" -otsv)
+for role in "Virtual Machine Contributor" "Network Contributor" "Managed Identity Operator"; do
+  az role assignment create --assignee "${KARPENTER_USER_ASSIGNED_CLIENT_ID}" --scope "${RG_MC_RES}" --role "$role"
+done
+```
+
+### Configure Helm chart values
+
+Karpeter Helm chart requires some configuration via values to work with a specific AKS cluster. The values are documented in the Helm chart itself, but you can use `configure-values.sh` to generate `karpenter-values.yaml` with the required configuration. The script interrogates the AKS cluster and generates the values file, using `karpenter-values-template.yaml` as a template.
```

Comment:
I would add a note that the script has a side effect of fetching the `karpenter-values-template.yaml` as well